detox/src/artifacts/screenshot/SimulatorScreenshotPlugin.js

```diff
@@ -51,19 +53,21 @@ class SimulatorScreenshotPlugin extends ScreenshotArtifactPlugin {
   }
 
   _onInvokeFailure({ params }) {
-    const artifacts = {
-      visibilityFailingScreenshots: params.visibilityFailingScreenshotsURL,
-      visibilityFailingRects: params.visibilityFailingRectsURL,
-    };
-
-    for (const [key, artifactPath] of Object.entries(artifacts)) {
-      if (!artifactPath || fs.isDirEmptySync(artifactPath)) {
-        continue;
+    const visibilityArtifactDirs = [
+      params.visibilityFailingScreenshotsURL,
+      params.visibilityFailingRectsURL
+    ].filter(Boolean);
+
+    for (const visiblityDir of visibilityArtifactDirs) {
+      for (const innerFile of fs.readdirSync(visiblityDir)) {
+        const ext = path.extname(innerFile);
+        if (ext === '.png') {
+          const artifactName = path.basename(innerFile, ext).replace(/ /g, '_');
+          this._registerSnapshot(artifactName, new FileArtifact({
+            temporaryPath: path.join(visiblityDir, innerFile),
+          }));
+        }
```

Comment:
'Really wish we'd have these things unit-tested. Both the logic and special-char replacement are a bit fancy.

Reply:
I partially addressed this in the E2E tests by asserting quite a strict file mask: https://github.com/wix/Detox/pull/3108/commits/3376f3ac391778f441e9a9d582cf9708b2d207cf

Indeed, the artifacts should undergo some rewrite. They smell like a legacy code for a while now.